copy.lua

```diff
@@ -0,0 +1,417 @@
+                    local cursor2=cursor
+                    if not (gui.blink_visible(750) and x==-offsetX and y==-offsetY) then
+                        local fg=COLOR_BLACK
+                        local bg=COLOR_CYAN
+                        if self.option=='erase' then
+                            bg=COLOR_RED
+                            fg=COLOR_BLACK
+                        elseif self.option=='construction' then
+                            bg=COLOR_GREEN
+                            fg=COLOR_BLACK
+                        end
+                        local symbol=digSymbols[tile.dig]
+                        if self.option~='normal' then
+                            symbol=' '
+                        end
+                        dc:pen(fg,bg)
+                        paintMapTile(dc, vp, cursor2, xyz2pos(df.global.cursor.x+x+offsetX,df.global.cursor.y+y+offsetY,df.global.cursor.z), symbol, fg)
+                    end
+                
+                    
+                    --
+                end
+            end
+        end
+        
+    end
+end
+
+function CopyUI:onRenderBody(dc)
+    self:renderOverlay(cursor)
+    
+
+    dc:clear():seek(1,1):pen(COLOR_WHITE):string("Copy UI - "..self.state:gsub("^%a",function(x)return x:upper()end))
+    dc:seek(2,3)
+    local cursor = guidm.getCursorPos()
+    
+    
+
+    
+    if self.state=='paste' then
+        dc:key('CUSTOM_S'):string(": Set Clipboard",COLOR_GREY)
+        
+--
+        dc:newline():newline(1)
+        dc:key('CUSTOM_H'):string(": Flip Horizontal",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_V'):string(": Flip Vertical",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_R'):string(": Rotate 90",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_T'):string(": Rotate -90",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_G'):string(": Cycle Corner",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_I'):string(": Invert",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_C'):string(": Convert to...",COLOR_GREY) dc:newline(1)
+        dc:newline(1)
+        dc:key('CUSTOM_E'):string(": "..(self.option=='erase' and "Erasing" or "Erase"),self.option=='erase' and COLOR_RED or COLOR_GREY) dc:newline(1) --make red
+        dc:key('CUSTOM_X'):string(": "..(self.option=='construction' and "Canceling Contructions" or "Cancel Constructions"),self.option=='construction' and COLOR_GREEN or COLOR_GREY) dc:newline(1) --make red
+        dc:newline():newline(1)
+        dc:key('CUSTOM_B'):string(": Blink Clipboard",self.blink and COLOR_WHITE or COLOR_GREY) dc:newline(1)
+        dc:newline()
+    elseif self.state=='mark' then
+        if self.buffer==nil then
+            dc:string("Select two corners.")
+        end
+        dc:newline():newline(1)
+        dc:key('CUSTOM_P'):string(": Cull Selections",self.cull and COLOR_WHITE or COLOR_GREY)
+    elseif self.state=='convert' then
+        dc:key('CUSTOM_D'):string(": Mine",COLOR_GREY) dc:newline(2)
```

Comment:
`dc:key_string('CUSTOM_D', 'Mine') is equivalent, and you should chain the `:newline()` instead of putting two statements on the same line, which is harder to read.